core/src/databases.rs

```diff
@@ -18,22 +18,50 @@ impl MongoDatabases {
     // All columns except the TABLE_CAT column contain NULLs.
     // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
     // timeout, the query must finish before the timeout or an error is returned.
-    pub fn list_all_catalogs(_client: &MongoConnection, _query_timeout: Option<i32>) -> Self {
-        unimplemented!()
+    pub fn list_all_catalogs(
+        mongo_connection: &MongoConnection,
+        _query_timeout: Option<i32>,
+    ) -> Self {
+        let database_names: Vec<String> = mongo_connection
+            .client
+            .list_database_names(None, None)
+            .unwrap();
+        let mut db_names: Vec<Bson> = Vec::new();
+        for name in database_names {
+            db_names.push(Bson::String(name));
+        }
+        MongoDatabases {
+            database_names: db_names,
+            current_db_index: 0,
+        }
     }
 }
 
 impl MongoStatement for MongoDatabases {
     // Increment current_db_index.
     // Return true if current_db_index index is < for databases_names.length.
     fn next(&mut self) -> Result<bool> {
-        unimplemented!()
+        let current = self.current_db_index;
+        self.current_db_index += 1;
+        Ok(current < u32::try_from(self.database_names.len()).unwrap())
```

Comment:
We might just as well store the Vector length in the struct and skip the conversion part done each time next is called. WDYT?

Reply:
Sounds good, added the vector length in the struct and used that here to avoid the conversion for each next.